pkg/config/remote/service/clients.go

```diff
@@ -23,29 +23,57 @@ type cacheBypassClients struct {
 	clock    clock.Clock
 	requests chan chan struct{}
 
+	capacity int
+
 	// Fixed window rate limiting
 	// It allows client requests spikes while limiting the global amount of request
-	currentWindow  time.Time
-	windowDuration time.Duration
-	capacity       int
-	allowance      int
+	mu struct {
```

Comment:
this struct is never called by multiple goroutines, while this adds close to no overhead it's not necessary